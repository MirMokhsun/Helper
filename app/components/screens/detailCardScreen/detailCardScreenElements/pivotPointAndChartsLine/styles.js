import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        flex: 1,
        justifyContent: 'space-around',
        alignItems: 'center',
    },
    text: {
        fontFamily: 'Roboto-Light',
        lineHeight: 27,
        fontSize: 14,
        color: '#000756',
        marginRight: 10,
    },
    piwotPointWrapper: {
        flex: 3,
        flexDirection: 'row',
        justifyContent: 'center',
    },
    toogleWrapper: {
        flex: 2,
        justifyContent: 'center',
    },
    containerPivotPoint: {
        flexDirection: 'row',
        height: 25,
        justifyContent: 'center',
    },
    whiteNumbers: {
        lineHeight: 25,
        fontSize: 20,
        color: "#9FA0B2",
        fontFamily: 'Roboto-Regular',
    },
    largeNumbers: {
        paddingTop: 0.5,
        fontSize: 23,
    },
    valuestyle: {

    },
})