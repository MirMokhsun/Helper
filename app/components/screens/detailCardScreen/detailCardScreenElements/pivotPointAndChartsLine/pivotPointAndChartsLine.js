import React, { PureComponent } from 'react';
import { View, Text } from 'react-native';
import { styles } from './styles';
import { ChartsToogle } from '../chartsToogle/ChartsToogle';
import { setCamelPos } from '../../../../../api/cardQuotation.api/cardquotation.api';

export class PivotPointAndChartsLine extends PureComponent {
    constructor(props) {
        super(props)
        this.countUperNambers = 2;
        this.rendersNumb = setCamelPos(props.pivotPoint, styles.whiteNumbers, this.countUperNambers, styles);
        this.state = {
            rendersNumb: this.rendersNumb,
        }
    }

    componentWillReceiveProps({ pivotPoint }) {
        const rendersNumb = setCamelPos(pivotPoint, styles.whiteNumbers, this.countUperNambers, styles);
        this.setState({
            rendersNumb,
        });
    }

    render() {
        const { changeCharts } = this.props;
        const { rendersNumb } = this.state;
        return (
            <View style={styles.container}>
                <View style={styles.piwotPointWrapper}>
                    <Text style={styles.text}>Pivot Points: { rendersNumb }</Text>
                </View>
                <View style={styles.toogleWrapper}>
                    <ChartsToogle changeCharts={changeCharts} />
                </View>
            </View >
        )
    }
}

export default PivotPointAndChartsLine;