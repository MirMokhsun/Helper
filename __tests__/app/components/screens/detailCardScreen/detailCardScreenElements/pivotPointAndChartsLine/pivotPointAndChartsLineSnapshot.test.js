import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { PivotPointAndChartsLine } from '../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/pivotPointAndChartsLine/pivotPointAndChartsLine';

it('create snapshot PivotPointAndChartsLine', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<PivotPointAndChartsLine pivotPoint={'0.981390'} />
    );
    expect(toolbarSnapshot).toMatchSnapshot();
});