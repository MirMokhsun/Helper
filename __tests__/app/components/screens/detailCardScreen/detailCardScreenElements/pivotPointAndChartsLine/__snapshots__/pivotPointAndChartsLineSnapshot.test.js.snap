// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot PivotPointAndChartsLine 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "flex": 3,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000756",
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "lineHeight": 27,
          "marginRight": 10,
        }
      }
    >
      Pivot Points: 
      <Text
        style={
          Array [
            Object {
              "color": "#9FA0B2",
              "fontFamily": "Roboto-Regular",
              "fontSize": 20,
              "lineHeight": 25,
            },
            Object {},
          ]
        }
      >
        0
        .
        9
        8
        1
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9FA0B2",
              "fontFamily": "Roboto-Regular",
              "fontSize": 20,
              "lineHeight": 25,
            },
            Object {
              "fontSize": 23,
              "paddingTop": 0.5,
            },
          ]
        }
      >
        3
        9
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#9FA0B2",
              "fontFamily": "Roboto-Regular",
              "fontSize": 20,
              "lineHeight": 25,
            },
            Object {},
          ]
        }
      >
        0
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <ChartsToogle />
  </View>
</View>
`;
